lib/note-detail/index.jsx

```diff
@@ -64,8 +65,11 @@ export class NoteDetail extends Component {
 
   isValidNote = note => note && note.id;
 
-  componentWillReceiveProps() {
-    this.queueNoteSave.flush();
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.note.id !== this.props.note.id) {
```

Comment:
can this be called when `note` is undefined.

Reply:
Most likely, it's JavaScript.